kuma/core/i18n.py

```diff
@@ -1,4 +1,165 @@
+"""
+Customizations of Django i18n functions for Kuma.
+
+Django language code is lower case, like 'en-us'.
+Kuma language code is mixed case, like 'en-US'.
+"""
 from django.apps import apps
+from django.conf import settings
+from django.conf.locale import LANG_INFO
+from django.utils import lru_cache, translation
+from django.utils.translation.trans_real import (
+    check_for_language, get_languages, language_code_prefix_re,
+    language_code_re, parse_accept_lang_header)
+
+
+def django_language_code_to_kuma(lang_code):
+    """
+    Convert Django language code to Kuma language code.
+
+    Django uses lower-case codes like en-us.
+    Mozilla uses mixed-case codes like en-US.
+    """
+    return settings.LANGUAGE_URL_MAP.get(lang_code, lang_code)
+
+
+def kuma_language_code_to_django(lang_code):
+    """
+    Convert Kuma language code to Django.
+
+    Django uses lower-case codes like en-us.
+    Mozilla uses mixed-case codes like en-US.
+    """
+    return lang_code.lower()
+
+
+def get_language():
+    """Returns the currently selected language as a Kuma language code."""
+    return django_language_code_to_kuma(translation.get_language())
+
+
+@lru_cache.lru_cache(maxsize=1000)
+def get_supported_language_variant(raw_lang_code):
+    """
+    Returns the language-code that's listed in supported languages, possibly
+    selecting a more generic variant. Raises LookupError if nothing found.
+
+    If `strict` is False (the default), the function will look for an alternative
+    country-specific variant when the currently checked is not found.
+
+    lru_cache should have a maxsize to prevent from memory exhaustion attacks,
+    as the provided language codes are taken from the HTTP request. See also
+    <https://www.djangoproject.com/weblog/2007/oct/26/security-fix/>.
+
+    Based on Django 1.8.19's get_supported_language_variant from
+    django/utils/translation/trans_real.py, with changes:
+
+    * Language code can also be a Kuma language code
+    * Return Kuma languge codes
+    * Force strict=False to always allow fuzzy matching (zh-CHS gets zh-CN)
+    """
+    if raw_lang_code:
+        lang_code = kuma_language_code_to_django(raw_lang_code)
+
+        # Check for known override
+        if lang_code in settings.LOCALE_ALIASES:
+            return settings.LOCALE_ALIASES[lang_code]
```

Comment:
❤️ I love that the `LOCALE_ALIASES` are incorporated into this here. They can be handled in a much cleaner, more straightforward fashion.

Reply:
Yes, it is a much better fit than the "fix after the fact" method from the last PR.